vpd-tool/src/vpd_tool_main.cpp

```diff
@@ -47,38 +48,46 @@ int main(int argc, char** argv)
 
     CLI11_PARSE(l_app, argc, argv);
 
-    if (*l_objectOption && l_vpdPath.empty())
+    if ((l_objectOption->count() > 0) && l_vpdPath.empty())
     {
         std::cout << "Given path is empty." << std::endl;
-        return -1;
+        return l_rc;
     }
 
-    if (*l_recordOption && (l_recordName.size() != vpd::constants::RECORD_SIZE))
+    if ((l_recordOption->count() > 0) &&
+        (l_recordName.size() != vpd::constants::RECORD_SIZE))
     {
         std::cerr << "Record " << l_recordName << " is not supported."
                   << std::endl;
+        return l_rc;
     }
 
-    if (*l_keywordOption &&
+    if ((l_keywordOption->count() > 0) &&
         (l_keywordName.size() != vpd::constants::KEYWORD_SIZE))
     {
         std::cerr << "Keyword " << l_keywordName << " is not supported."
                   << std::endl;
         return l_rc;
     }
 
-    if (*l_hardwareFlag && !std::filesystem::exists(l_vpdPath))
-    {
-        std::cerr << "Given EEPROM file path doesn't exist : " + l_vpdPath
-                  << std::endl;
-        return l_rc;
-    }
-
     (void)l_fileOption;
 
-    if (*l_readFlag)
+    if (l_readFlag->count() > 0)
```

Comment:
if user provides multiple -r in the command line, we should throw error.
so please check if the l_hardwareFlag->count() == 1 to proceed. Else throw unexpected command line option

Reply:
If the user mistakenly gives twice, this is still fine.

we are taking action as user chooses the option, instead of blocking the results or throwing an error.

Reply:
To resolve this, should we check here how the standard command line operations work?

Reply:
yes, I checked few linux commands, it works with duplicate options given.

`anupama@gfwa601:~$ ls -lh -lh
total 1.5G
-rw-r--r--  1 anupama 1122842 154K Nov  7 10:21 50001001.json
-rwxr-xr-x  1 anupama 1122842  17K Nov 12 02:39 a.out
drwxr-xr-x  2 root    root    4.0K Jun 28 01:54 anupama_san5

anupama@gfwa601:~$ grep -nr -r "BUS" test/*
grep: test/factory.hpp.gch: binary file matches`

Reply:
In that case I am ok with your code.